src/EventStoreIntegration/AggregateTranslator.php

```diff
@@ -68,6 +58,15 @@ public function extractPendingStreamEvents($anEventSourcedAggregateRoot)
     }
 
     /**
+     * @param object $anEventSourcedAggregateRoot
+     * @param Message[] $events
+     */
+    public function applyPendingStreamEvents($anEventSourcedAggregateRoot, array $events)
```

Comment:
This is the only method that wants an array instead of an iterator. What would you think of making this consistent with the other methods?

To achieve this, we would do in the AggregateRootDecorator this:

    public function extractRecordedEvents(AggregateRoot $anAggregateRoot)
    {
        return new \ArrayIterator($anAggregateRoot->popRecordedEvents());
    }

Also update the ConfigurableAggregateTranslator and the AggregateRepository classes.

On the other hand, it' not really needed, but the methods here would have consistent method headers.